### DISCUSSION TITLE
Stuck on BIOS ReBar Mod - Unable to fix File padding issue caused by the tools or code required by steps of the guide.

### DISCUSSION THREAD

**Discussion by SlowerBeen** [xCuri0/ReBarUEFI#161] [Q&A]
Hi there,
Would be great to have a full explanation how do i fix the file padding issue on my maximus 5 gene bios, i have enabled 4g decode on bios native by AMIBCP i have the NVEspress5 module for NVME Support, and i am able to insert the Rebar Module Mod.

The only thing is not working is the patching step of the bios by UEFIpatch witch causes pading issue , even when i use MMtool for replace the patched modules, padding issue occurs.

Please some one give me some more information, is it not clear about what is causing this issue and how to solve by the information that i found here-,

**Comment by ZOXZX** [comment]
You can try removing OS entries as in DSDT troubleshooting guide, or changing values manually with HxD ...

**Reply by SlowerBeen** [reply]
Thanks for your answer, could you please explain what you mean whit removing OS entries ? i havent got to DSDT step due to this padding file issue, also could you please explain me the procedure to find and fix the HEX Values ? i allready have the Hex editor installed.

**Reply by ZOXZX** [reply]
Eh, sorry about DSDT, that is the next step if needed so not applicable here.
Look at what UEFIpatch is doing, search for that values in original and modded bios and replace bytes by hex editor.
After that, fix checksum errors by finding header offset with UEFItoolNE, going there with HxD and change checksum to one displayed by NE...

**Reply by SlowerBeen** [reply]
(I dont know why i can see emails but i canot se our previous conversation here, i hope is not some moderator messing around in my life business ;)  )

Any way, i was able to patch a bios , extract the patched modules whit and replace them to unpatched bios whit MM tool, now my system booted even whit file "pad issue" and UEFI patch says nothing to patch on the MM tool patched bios.

I dont know why this lady was able to build this mod and not being able to write a non weird guide.

Now would be very help full if you can explain me how to apply the DSTD patch without getting errors, i get 6 external unresolved and when i try to recompile i get 2 errors and canot complete compiling.

(hex editor solution would have been a "brain cells killing solution")

**Reply by ZOXZX** [reply]
Guide is clear if you follow it step by step, but I am confused now: first you said that you can't replace patched modules but now that there is nothing to patch. I am assuming you just inserted modules into original bios ...
Anyway, padding issue can represent in numerous ways not all of which are visible right away: e.g. unstable system, incorrect memory speeds, etc.
Hex replacing is very easy since you know what to search and replace:
https://github.com/xCuri0/ReBarUEFI/issues/48#issuecomment-1890331162

Did you go over everything?
https://github.com/xCuri0/ReBarUEFI/wiki/Common-issues-(and-fixes)
https://github.com/xCuri0/ReBarUEFI/wiki/Using-UEFIPatch#pad-file-issue-workaround
https://github.com/xCuri0/ReBarUEFI/wiki/DSDT-Patching#sandyivy-bridge-dsdt-patch

Edit:
You can go over the second part of this guide (although it's for x79) to see how to do hex replacement:
https://www.youtube.com/watch?v=78sdn_H5-pM&t=996s

**Reply by SlowerBeen** [reply]
the last one would be helpfull (i hope so) , any way now i would like to test my system, i only need dstd patching
my system is working now.
i will try that hex fix only if i will have padding related problems 

above 4g encoding enabled, (enable native with AMIBCB supervisor setting, the tool provided in the guide for enabling 4g hident setting did nothing, so whats the point of using a non official tool without first suggesting to use of an official one ? , it was so easy to do it) csm disabled,  RebarModule installed + patches, but i am having problems decompiling dstd and compiling it back witch of course, no details are provided for those problems.

i have followed the guide step by step but those steps didnt work has advertised i had to figure out solution or find ones around the web, just like you did now, sending me a guide for x79 witch actualy has information for general problem solving of this mod.



it is not clear endeed , thank to you that are helping me.

**Reply by ZOXZX** [reply]
There are no unofficial version AFAIK, but some versions may display different menu structure.
AMIBCP has two entries: failsafe and optimal. 
If you change optimal then you have to load optimized defaults in bios for them to be active. 
On most boards, changing visibility values does nothing so you have to change optimal settings for all of them to be loaded.
Don't touch failsafe values if you want to boot into bios if something goes wrong. They are loaded after flash and clear cmos.

List your DSDT problems so we can try to find a solution ...

**Reply by SlowerBeen** [reply]
maximus v gene above 4 gb visible in bios no problem, dxe driver from https://github.com/terminatorul/NvStrapsReBar

I have a gtx 980 i am hopping there is a way to activare rebar even if is so old, i dont beleive it is not compatible, it is just a mater pf software


i tried to activate 512 mb bar on but my system boots whit no screen signal, i dont know if it is because my gpu is not compatible or because i am missing the DSDT patch. any way 
this is what happens when i try to decompile DSDT
 * Intel ACPI Component Architecture
 * AML/ASL+ Disassembler version 20230628 (32-bit version)
 * Copyright (c) 2000 - 2023 Intel Corporation
 * 
 * Disassembling to symbolic ASL+ operators
 *
 * Disassembly of DSDT.aml, Sun Feb 25 12:29:21 2024
 *
 * Original Table Header:
 *     Signature        "DSDT"
 *     Length           0x0000AECE (44750)
 *     Revision         0x02
 *     Checksum         0x0A
 *     OEM ID           "ALASKA"
 *     OEM Table ID     "A M I"
 *     OEM Revision     0x00000022 (34)
 *     Compiler ID      "INTL"
 *     Compiler Version 0x20051117 (537202967)
 */
DefinitionBlock ("", "DSDT", 2, "ALASKA", "A M I", 0x00000022)
{
    /*
     * iASL Warning: There were 6 external control methods found during
     * disassembly, but only 0 were resolved (6 unresolved). Additional
     * ACPI tables may be required to properly disassemble the code. This
     * resulting disassembler output file may not compile because the
     * disassembler did not know how many arguments to assign to the
     * unresolved methods. Note: SSDTs can be dynamically loaded at
     * runtime and may or may not be available via the host OS.
     *
     * To specify the tables needed to resolve external control method
     * references, the -e option can be used to specify the filenames.
     * Example iASL invocations:
     *     iasl -e ssdt1.aml ssdt2.aml ssdt3.aml -d dsdt.aml
     *     iasl -e dsdt.aml ssdt2.aml -d ssdt1.aml
     *     iasl -e ssdt*.aml -d dsdt.aml
     *
     * In addition, the -fe option can be used to specify a file containing
     * control method external declarations with the associated method
     * argument counts. Each line of the file must be of the form:
     *     External (<method pathname>, MethodObj, <argument count>)
     * Invocation:
     *     iasl -fe refs.txt -d dsdt.aml
     *
     * The following methods were unresolved and many not compile properly
     * because the disassembler had to guess at the number of arguments
     * required for each:
     */
    External (_PR_.AAC0, UnknownObj)
    External (_PR_.ACRT, UnknownObj)
    External (_PR_.APSV, UnknownObj)
    External (_PR_.CPU0._PPC, UnknownObj)
    External (_SB_.PCI0.IEIT.EITV, MethodObj)    // Warning: Unknown method, guessing 0 arguments
    External (CFGD, UnknownObj)
    External (ECST, MethodObj)    // Warning: Unknown method, guessing 1 arguments
    External (HDOS, MethodObj)    // Warning: Unknown method, guessing 0 arguments
    External (HNOT, MethodObj)    // Warning: Unknown method, guessing 1 arguments
    External (HWID, IntObj)
    External (IDAB, MethodObj)    // Warning: Unknown method, guessing 0 arguments
    External (PDC0, UnknownObj)
    External (PDC1, UnknownObj)
    External (PDC2, UnknownObj)
    External (PDC3, UnknownObj)
    External (PDC4, UnknownObj)
    External (PDC5, UnknownObj)
    External (PDC6, UnknownObj)
    External (PDC7, UnknownObj)
    External (TNOT, MethodObj)    // Warning: Unknown method, guessing 0 argume

**Reply by ZOXZX** [reply]
Sorry, that is above my knowledge ...

For DSDT, something similar is here:
https://www.insanelymac.com/forum/topic/301908-why-you-need-proper-aml-disassembly/

"The solution is simple because it comes with the iASL warning above.
iasl -e <ssdt1.aml,ssdt2.aml...> -d <dsdt.aml>

or for dissembling all files at once:
iasl -da DSDT.aml SSDT*.aml 

If you do it like advised, you don't get this kind of errors and the method looks like that:

        Method (ADBG, 1, Serialized)
  
        If (CondRefOf (MDBG))
        {
            Return (MDBG (Arg0))
        }
        Return (Zero)
  
If I do this for my Gigabyte H81M-HD3, I got only the well known "Name already exits in this scope..." errors.
But IHMO, these are homemade (iASL) errors because iASL puts the unresolved methods at the top of the .dsl file.
If I put them back to the end of the first block where all external methods are defined - all errors are gone.
No other fixes where needed to compile my DSDT."

You'll need someone more knowledgeable to fix 980 rebar.
I see it uses ffs from other project and I can't follow all changes and methods since I don't have nVidia card.
